### PR TITLE
Removing restart logic

### PR DESCRIPTION
**Pull request by RodneyU215**
### Summary
1. The method `#restart!` has been removed from `SlackRubyBot::Server`. It was causing issues with the restart logic that was recently added to [`slack-ruby-client`](https://github.com/slack-ruby/slack-ruby-client/pull/226). 

### Requirements
* [x] I've read and understood the [Contributing Guidelines](https://github.com/slack-ruby/slack-ruby-bot/blob/master/CONTRIBUTING.md) and have done my best effort to follow them.

### CODE REVIEW

**Comment by dblock** [comment]
Lets clarify UPGRADING, make a major version bump, and lock the version of slack-ruby-bot to the next version of slack-ruby-client in .gemspec part of this PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 * [#198](https://github.com/slack-ruby/slack-ruby-bot/pull/198): Recommend using `async-websocket` instead of `celluloid-io` - [@dblock](https://github.com/dblock).
 * [#202](https://github.com/slack-ruby/slack-ruby-bot/pull/202): Allow frozen string in Hooks::Message - [@jonosenior](https://github.com/jonosenior).
+* * [#203](https://github.com/slack-ruby/slack-ruby-bot/pull/203): Removing restart logic - [@RodneyU215](https://github.com/RodneyU215).
```
Comment (dblock) [inline CHANGELOG.md]: Double **

**Comment by RodneyU215** [comment]
@dblock I've addressed the feedback. Please let me know how you'd like to move forward. If my understanding is correct, these changes won't be compatible until we release a new version of `slack-ruby-client`.

File: UPGRADING.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
 Upgrading SlackRubyBot
 ======================
 
+### Upgrading to >= 0.12.0
+
+#### Remove any references to `SlackRubyBot::Server#restart!`
+
+We have deprecated `SlackRubyBot::Server#restart!` since the [`slack-ruby-client`](https://github.com/slack-ruby/slack-ruby-client/blob/master/CHANGELOG.md) now restarts any connection that was not intentionally stopped via `SlackRubyBot::Server#stop!`.
```
Comment (dblock) [inline UPGRADING.md]: deprecated -> removed

**Comment by RodneyU215** [comment]
It's finally ready! 🎉😅

**Comment by dblock** [comment]
I'll keep this open for now till we can at least feature complete https://github.com/slack-ruby/slack-ruby-client/issues/223 for other async IO implementations.

**Comment by dblock** [comment]
Care to rebase this @RodneyU215 ? I'll merge.

**Pull request merged**